server/vulnerabilities/macoffice/analyzer.go

```diff
@@ -101,26 +101,28 @@ func getStoredVulnerabilities(
 
 func updateVulnsInDB(
 	ctx context.Context,
-	ds fleet.Datastore,
 	detected []fleet.SoftwareVulnerability,
 	existing []fleet.SoftwareVulnerability,
+	delVulnsDBCall func(ctx context.Context, vulnerabilities []fleet.SoftwareVulnerability) error,
+	insertVulnsDBCall func(ctx context.Context, vulns []fleet.SoftwareVulnerability, source fleet.VulnerabilitySource) (int64, error),
```

Comment:
How come you are changing the function signature but I can't see the change for the code that calls it?